js/blockchain.js

```diff
@@ -27,205 +28,500 @@ if (typeof XPCNativeWrapper === 'function') {
+        if (obj.headers) {
+            Object.assign(obj.headers, addtlHeaders);
+        } else {
+            obj.headers = addtlHeaders;
         }
-    }).done(function(response) {
-        //console.log(response);
-        queuedStorage[user_name] = {type: connectionType, connection: currentProfileName, on: Date.now(), id: String(user_id)};
-    }).fail(function(xhr, text, err) {
-        errors++;
-        $("#blockchain-dialog .errorCount").text(errors);
-        //console.log(xhr);
-    }).always(function() {
+        return new Promise((resolve, reject) => {
+            let xhr = new XMLHttpRequest();
+            xhr.open(obj.method || "GET", obj.url);
+            if (obj.headers) {
+                Object.keys(obj.headers).forEach(key => {
+                    xhr.setRequestHeader(key, obj.headers[key]);
+                });
+            }
+            xhr.onload = () => {
+                if (xhr.status >= 200 && xhr.status < 300) {
+                    resolve(xhr.response);
+                } else {
+                    reject(xhr.statusText);
+                }
+            };
+            xhr.onerror = () => reject(xhr.statusText);
+            xhr.send(obj.body);
+        });
+    }
+    startAccountFinder() {
+        finderRunning = true;
+        const profileUsername = currentProfileName;
+        const requestType = window.location.href.split("/").pop();
+        let position = $(".GridTimeline-items").data('min-position');
+        let lastRequestTime = Date.now();
+        let cursor = null;
+        const _getIDData = () => {
+            if (!finderRunning) return false;
+            const count = 5000;
+            let url = `https://api.twitter.com/1.1/${requestType}/ids.json?screen_name=${profileUsername}&count=${count}&stringify_ids=true`
+            if (cursor) url += `&cursor=${cursor}`
+            lastRequestTime = Date.now();
+            return this._makeRequest({
+                url: url,
+                method: 'GET'
+            }).then((response) => {
+                let jsonData = JSON.parse(response);
+                if (jsonData.hasOwnProperty('next_cursor_str') && jsonData.next_cursor_str != "0") {
+                    cursor = jsonData.next_cursor_str;
+                } else {
+                    cursor = null;
+                }
+                return jsonData;
+            });
+        }
+        const _getUserData = (jsonData) => {
+            if (jsonData.hasOwnProperty('ids')) {
+                // split array of ids into 100 user chunks
+                let i = 0;
+                let chunks = [];
+                while (i * 100 < jsonData.ids.length) {
+                    let slice = jsonData.ids.slice(i * 100, i * 100 + 100);
+                    chunks.push(slice.join(","));
+                    i++;
+                }
+                chunks = chunks.map((element) => {
+                    let url = 'https://api.twitter.com/1.1/users/lookup.json?include_entities=true'
```

Comment:
in `_processData` function, `element` missing `.blocking` property. so `url` in here should include `include_blocking=true`. (note that this is undocumented parameter)
```diff
- let url = 'https://api.twitter.com/1.1/users/lookup.json?include_entities=true'
+ let url = 'https://api.twitter.com/1.1/users/lookup.json?include_entities=true&include_blocking=true'
```

Reply:
nice find! thank you. added this, also remove the code that hid the "already blocking" field in the dialog.